pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook_test.go

```diff
@@ -0,0 +1,381 @@
+package fleetresourcehandler
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+	"github.com/stretchr/testify/assert"
+	admissionv1 "k8s.io/api/admission/v1"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	fleetv1alpha1 "go.goms.io/fleet/apis/v1alpha1"
+	"go.goms.io/fleet/pkg/utils"
+)
+
+const (
+	resourceAllowedFormat                = "user: %s in groups: %v is allowed to modify resource %s: %+v"
+	resourceDeniedFormat                 = "user: %s in groups: %v is not allowed to modify resource %s: %+v"
+	resourceStatusUpdateNotAllowedFormat = "user: %s in groups: %v is not allowed to update %s status: %+v"
+	resourceAllowedGetMCFailed           = "user: %s in groups: %v is allowed to updated %s: %+v because we failed to get MC"
+)
+
+func TestHandleInternalMemberCluster(t *testing.T) {
+	mockClient := &test.MockClient{
```

Comment:
why not make this a field so we can test failure cases?

Reply:
The mock client is being passed as a field within the request object and I declared it here to reduce LOC where I saw the same client is being reused and I am testing a failure case as well